libs/template/config.go

```diff
@@ -92,63 +92,102 @@ func (c *config) assignDefaultValues(r *renderer) error {
 	return nil
 }
 
-// Prompts user for values for properties that do not have a value set yet
-func (c *config) promptForValues(r *renderer) error {
-	for _, p := range c.schema.OrderedProperties() {
-		name := p.Name
-		property := p.Schema
+func (c *config) promptSelect(property jsonschema.Property, r *renderer) error {
+	name := property.Name
+	schema := property.Schema
 
-		// Config already has a value assigned
-		if _, ok := c.values[name]; ok {
-			continue
-		}
+	// Compute description
+	description, err := r.executeTemplate(schema.Description)
+	if err != nil {
+		return err
+	}
 
-		// Compute default value to display by converting it to a string
-		var defaultVal string
-		var err error
-		if property.Default != nil {
-			defaultValRaw, err := property.DefaultString()
-			if err != nil {
-				return err
-			}
-			defaultVal, err = r.executeTemplate(defaultValRaw)
-			if err != nil {
-				return err
-			}
-		}
+	// List of options to display to user
+	options, err := schema.EnumStringSlice()
+	if err != nil {
+		return err
+	}
+
+	// Get user input
+	userInput, err := cmdio.AskSelect(c.ctx, description, options)
+	if err != nil {
+		return err
+	}
+
+	// Convert user input string back to a value
+	c.values[name], err = schema.ParseString(userInput)
+	if err != nil {
+		return err
+	}
+
+	// Validate the partial config based on this update
+	return c.schema.ValidateInstance(c.values)
+}
+
+func (c *config) promptText(property jsonschema.Property, r *renderer, numRetries int) error {
+	name := property.Name
+	schema := property.Schema
+
+	// Compute description
+	description, err := r.executeTemplate(schema.Description)
+	if err != nil {
+		return err
+	}
+
+	// Compute default value
+	defaultRaw, err := schema.DefaultString()
+	if err != nil {
+		return err
+	}
+	defaultVal, err := r.executeTemplate(defaultRaw)
+	if err != nil {
+		return err
+	}
 
-		description, err := r.executeTemplate(property.Description)
+	for i := 0; i < numRetries; i++ {
```

Comment:
Why limit retries? User can always Ctrl-C out of it if they're fed up.

Reply:
Done. We shall bring the power of infinite retries to our customers.